[WPE] WPE Platform: add support for loading external platforms

carlosgcampos opened the pull request:
#### d8b7a2f7b925af8627dd15cef0aec0e68af31f0c
<pre>
[WPE] WPE Platform: add support for loading external platforms
<a href="https://bugs.webkit.org/show_bug.cgi?id=265641">https://bugs.webkit.org/show_bug.cgi?id=265641</a>

Reviewed by Adrian Perez de Castro.

* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe-platform.pc.in:
* Source/WebKit/WPEPlatform/wpe/WPEExtensions.cpp:
(wpeEnsureExtensionPointsLoaded):

Canonical link: <a href="https://commits.webkit.org/279640@main">https://commits.webkit.org/279640@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1ac6c7d40e356462bd7304c06eac70a3a4924a95

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54071 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33450 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6603 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57346 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4795 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40968 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4688 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43785 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3179 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56167 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31679 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46803 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24927 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28501 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2944 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/50200 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4321 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58940 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29263 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/4425 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51200 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30441 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46918 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50557 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31401 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8007 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30223 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/WPEPlatform/wpe/WPEExtensions.cpp:
```diff
@@ -57,7 +58,16 @@ void wpeEnsureExtensionPointsLoaded()
+        const char* path = g_getenv("WPE_PLATFORMS_PATH");
+        if (path && *path) {
+            GUniquePtr<char*> paths(g_strsplit(path, G_SEARCHPATH_SEPARATOR_S, 0));
+            for (int i = 0; paths.get()[i]; ++i)
```
Array indexes cannot be negative, use an unsigned type:

```suggestion
            for (size_t i = 0; paths.get()[i]; ++i)
```

Also, we may want to check each path here with `g_file_query_file_type() == G_FILE_TYPE_DIRECTORY` before scanning it.

carlosgcampos replied:
hmm, first thing `g_io_modules_scan_all_in_directory_with_scope` does is `g_dir_open` and returns early when it fails.

aperezdc replied:
Nice, then it's less work to do o our end 👏🏼

aperezdc reviewed (approved):
LGTM with a nit.

carlosgcampos committed:
[WPE] WPE Platform: add support for loading external platforms
https://bugs.webkit.org/show_bug.cgi?id=265641

Reviewed by Adrian Perez de Castro.

* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe-platform.pc.in:
* Source/WebKit/WPEPlatform/wpe/WPEExtensions.cpp:
(wpeEnsureExtensionPointsLoaded):

Canonical link: https://commits.webkit.org/279640@main
